src/d3a/models/strategy/infinite_bus.py

```diff
@@ -15,29 +15,46 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-
+import d3a.constants
+from d3a.d3a_core.exceptions import MarketException
+from d3a.models.strategy import BidEnabledStrategy, INF_ENERGY
+from d3a.models.strategy.commercial_producer import CommercialStrategy
 from d3a_interface.constants_limits import ConstSettings, GlobalConfig
-from d3a_interface.utils import convert_str_to_pendulum_in_dict, convert_pendulum_to_str_in_dict
 from d3a_interface.read_user_profile import read_arbitrary_profile, InputProfileTypes, \
-    read_and_convert_identity_profile_to_float
-from d3a.models.strategy.commercial_producer import CommercialStrategy
-from d3a.models.strategy import BidEnabledStrategy, INF_ENERGY
-from d3a.d3a_core.exceptions import MarketException
+    read_and_convert_identity_profile_to_float, convert_identity_profile_to_float
+from d3a_interface.utils import convert_str_to_pendulum_in_dict, convert_pendulum_to_str_in_dict
 from d3a_interface.utils import find_object_of_same_weekday_and_time
 
 
 class InfiniteBusStrategy(CommercialStrategy, BidEnabledStrategy):
-    parameters = ('energy_sell_rate', 'energy_rate_profile', 'energy_buy_rate',
-                  'buying_rate_profile')
+    parameters = ("energy_sell_rate", "energy_rate_profile", "energy_buy_rate",
+                  "buying_rate_profile", "buying_rate_profile_uuid", "energy_rate_profile_uuid")
 
     def __init__(self, energy_sell_rate=None, energy_rate_profile=None, energy_buy_rate=None,
-                 buying_rate_profile=None):
+                 buying_rate_profile=None, buying_rate_profile_uuid=None,
+                 energy_rate_profile_uuid=None):
         super().__init__()
         self.energy_per_slot_kWh = INF_ENERGY
-        self.energy_buy_rate = energy_buy_rate
-        self.buying_rate_profile = buying_rate_profile
-        self.energy_rate = energy_sell_rate
-        self.energy_rate_profile = energy_rate_profile
+        self.energy_buy_rate = None
+
+        if d3a.constants.CONNECT_TO_PROFILES_DB and buying_rate_profile_uuid:
```

Comment:
In order to not have if checks for this constant throughout the code, I think that a function called `should_read_profile_from_db(profile_uuid)` would be much better. It also has the advantage that it can check the cornercase where the profile_uuid does not exist in the DB.